### ISSUE TITLE
Cost with obj

### ISSUE THREAD

**Issue by amir-zeldes** [UniversalDependencies/UD_English-EWT#505]
Currently "cost" takes `obj` in EWT (plus an error or two with `obl`):

https://universal.grew.fr/?custom=659ac9f55565b

In GUM it's mixed between `obj` and `obl:npmod`. I think the latter is correct because it's not really passivizable, so it's more like an extent adverbial IMO (similar to "wait"). Shall we consolidate to `obl:npmod`?

**Comment by nschneid** [comment]
I feel like this came up in the discussion of indirect objects...would you say "me" is a *direct* object in "It cost me $10"?

**Comment by nschneid** [comment]
And is it relevant that *cost* can license *anything*: "Does it cost anything to attend?" I'm not sure I can think of another place where an indefinite pronoun (as opposed to adjective or adverb) can be `obl:npmod`.

**Comment by nschneid** [comment]
I found this note from the iobj discussion:

> English has a few “quadrivalent” verbs, like _bet_ (CGEL, p. 219):
>
> - I bet you/iobj $10/obj [that it rains]/ccomp.
> 
> Note that _$10_ is an object here, or with _cost_, per the diagnostic that _what_ questions target subjects/objects, not adjuncts/adverbials (CGEL, p. 224): “What did you bet him that it would rain?” “What did it cost to repave the sidewalk?” (Also, _cost_ usually cannot be intransitive: *_The raincoat costs_. Exceptions occur with the sense of ‘take a toll’, typically for more abstract subjects.)

**Comment by amir-zeldes** [comment]
> would you say "me" is a direct object in "It cost me $10"

If we rule out $10 as an object then yeah, it would be the remaining object. FWIW, in IE languages with case marking, the "me" argument receives accusative case (e.g. in German, Polish), and these languages usually use dative for recipients. That suggests that a reading of "me" as a primary object is not so strange. The money argument is depreled as `obl` in UD Polish, and in German both arguments are `obj` (even concurrently in HDT), so probably one of them should have been iobj.

> anything: "Does it cost anything to attend?" I'm not sure I can think of another place where an indefinite pronoun can be obl:npmod.

I think it works with other measures, but usually modified:

  * I waited anything between 2 and 3 hours

The bigger issue is passivization IMO: I think you can't (normally?) say "$10 were costed by this purchase".

> Anyway, it seems that while "cost" is a weird verb, within English linguistics the passive construction isn't regarded as the be-all, end-all test for object. 

I definitely agree it's a weird verb. But I think passivization has been pretty central to our discussions of `obj` for UD English at least, and probably for other languages too. This is in the same ballpark as "wait" & co., so I think it's important to make a consistent decision. I'm not bent on it being `obl:npmod` and I could probably live with `obj` too, but I'd like to hear more opinions and make the data consistent in the end.

**Comment by nschneid** [comment]
> * I waited anything between 2 and 3 hours

Agreed that "anything" would occur with a temporal measure (in this example I'd personally prefer "anywhere" over "anything", but they're similar). Without a temporal measure it sounds strange:

- *Did you wait anything before the food arrived?

I think "anything" is actually functioning like a degree modifier of the PP. So not anything like an object. — Hey, there's another example of how it can modify a PP. ;)

**Comment by arademaker** [comment]
> And is it relevant that _cost_ can license _anything_: "Does it cost anything to attend?" I'm not sure I can think of another place where an indefinite pronoun can be `obl:npmod`.

Interesting that you take _anything_ as indefinite pronoun, for ERG it is analysed as any/quantifier and thing the second complement of the verb:

http://delph-in.github.io/delphin-viz/demo/#input=Does%20it%20cost%20anything%20to%20attend?&count=5&grammar=erg2018-uw&tree=true&mrs=true&dmrs=true

**Comment by nschneid** [comment]
Yes the compounding that gave rise to "anything" is transparent but treating it as a pronoun is longstanding UD policy. See e.g. [PRON](https://universaldependencies.org/u/pos/PRON.html). (Not sure that affects the criteria for `obj` though.)

**Comment by amir-zeldes** [comment]
> I think "anything" is actually functioning like a degree modifier of the PP

OK, maybe that's fine, but I'm not sure it matters for whether or not "cost" has a money obj or an extent modifier... And it seems UD languages have been handling it differently (e.g. Polish vs. German)
